tests/taskcluster/lib/decorate_task.js

```diff
@@ -96,12 +96,33 @@ function decorateTask(task, options) {
   output.task.scopes.push('docker-worker:image:' + IMAGE + '*');
 
   if (process.env.TREEHERDER_PROJECT && process.env.TREEHERDER_REVISION) {
+    // defaults to 'tc-treeherder' only so existing tasks need not be changed
+    var prefixes = ['tc-treeherder.'];
+
+    if (output.task['extra']) {
+      if (output.task.extra['treeherderEnv']) {
```

Comment:
hmm... I have mixed feeling about adding this to `task.extra`, because people could get the idea that `task.extra.treeherderEnv` dictates anything. Which it doesn't because it doesn't have any run-time effect. It's not read by the treeherder integration bits.

An alternative would be to add `task.treeherderEnv` and then delete it from the task object in the `decorateTask` method. On the other hand that could also be confusing... he he :)

So maybe just r+ and then hopefully we'll get around to come up with a better in-tree task configuration model. I know we're going to refactor scheduling eventually (hopefully I'll do it this quarter). Anyways, for this reason I don't think we need to scared of imperfect changes here...